File: Data Structures & Algorithms/is-palindrome/submission-2.js
class Solution {
 isPalindrome(s) {
       let cleanedStr = s.replace(/[^a-zA-z0-9]/gi, '').toLowerCase();
       let left = 0;
       let right = cleanedStr.length -1;

       while(left < right) {
        if(cleanedStr[left] !== cleanedStr[right]){
           return false; 
        }
            left++;
            right--;
       }
       return true;
    }
}

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-2.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int left = 0;
        int right = 0;
        int minLen = Integer.MAX_VALUE;
        int sum = 0;

        while(right < nums.length) {
            sum += nums[right];
            while(sum >= target) {
                minLen = Math.min(minLen, right - left + 1);
                sum -= nums[left];
                left++;
            }
            right++;
        }
        return minLen == Integer.MAX_VALUE ? 0 : minLen;
    }
}